File: frontend/src/api/generate.ts
import http from './http';
import { GenerateParams } from '../types';

export const createGenerateTask = async (data: {
  imageIds: string[];
  modelName: string;
  prompt: string;
  params: GenerateParams;
}) => {
  const response = await http.post('/api/generate', {
    ...data,
    mode: 'image',
  });
  return response.data;
};

export const getTaskStatus = async (taskId: string) => {
  const response = await http.get(`/api/generate/${taskId}`);
  return response.data;
};

File: backend/app/schemas/generate.py
from pydantic import BaseModel
from typing import List, Optional, Dict, Any


class GenerateParams(BaseModel):
    ratio: str = "1:1"
    count: int = 1


class GenerateRequest(BaseModel):
    imageIds: List[str]
    mode: str = "image"
    modelName: str
    prompt: str
    params: GenerateParams


class GenerateResponse(BaseModel):
    taskId: str


class ResultImage(BaseModel):
    imageId: str
    url: str


class TaskStatusResponse(BaseModel):
    taskId: str
    status: str
    progress: int
    resultImages: List[ResultImage]
    error: Optional[str] = None
